test/test-template-engine.js

```diff
@@ -0,0 +1,671 @@
+import XElement from '../x-element.js';
```

Comment:
New set of tests to spec out the functionality we want from our default template engine.